python_modules/dagster-externals/dagster_externals/_io/base.py

```diff
@@ -37,3 +42,58 @@ def load_context_params(self) -> ExternalExecutionParams:
     @abstractmethod
     def load_messages_params(self) -> ExternalExecutionParams:
         ...
+
+
+class ExternalExecutionBlobStoreMessageWriter(ExternalExecutionMessageWriter):
+    interval: float
+    max_chunk_size: int
+    buffer: List[ExternalExecutionMessage]
+    counter: int
+
+    def __init__(self, *, interval: float = 10):
+        self.interval = interval
+        self.buffer = []
+        self.counter = 1
+
+    @contextmanager
+    def setup(self, params: ExternalExecutionParams) -> Iterator[None]:
+        self.set_params(params)
+        thread = None
+        is_task_complete = Event()
+        try:
+            thread = Thread(target=self.writer_thread, args=(is_task_complete,), daemon=True)
```

Comment:
likely applies to existing subprocess impl - but if were going to use threads well need to either try to make them fork safe or explicitly decided that forking is not supported

Reply:
Is forking expected to occur somewhere? Or is this just in case the external script does it.

>likely applies to existing subprocess impl

When using file message reader/writer (default for subprocess) the threading is only on the orch side-- is it still a concern then?

Reply:
ya just concerned about the existing external process being something that forks it self. Not sure its likely but i think the spirit of this externals stuff is to bring as few constraints as possible so wanted to surface